tldr/semantic.py

```
@@ -1145,33 +1151,66 @@ def semantic_search(
+    search_k = min(k * 3, len(units)) if lang_extensions else min(k, len(units))
+
+    prev_candidate_count = 0
+    while True:
+        scores, indices = index.search(query_embedding, search_k)
+
+        # Count raw candidates (valid indices from FAISS) for stale-check
+        candidate_count = sum(1 for idx in indices[0] if 0 <= idx < len(units))
+
+        # Build results
+        results = []
+        for i, (score, idx) in enumerate(zip(scores[0], indices[0])):
```

Ruff (0.15.5)

[warning] Loop control variable `i` not used within loop body

Rename unused `i` to `_i`

(B007)